cf-revision-manager.php

```diff
@@ -3,199 +3,208 @@
 Plugin Name: CF Revision Manager
 Plugin URI: http://crowdfavorite.com
 Description: Revision management functionality so that plugins can add metadata to revisions as well as restore that metadata from revisions
-Version: 1.0.2
+Version: 2.0.0
 Author: Crowd Favorite
-Author URI: http://crowdfavorite.com 
+Author URI: http://crowdfavorite.com
 */
-if (!class_exists('cf_revisions')) {
-	
-	define('CF_REVISIONS_DEBUG', false);
-	
-	function cfr_register_metadata($postmeta_key, $display_func = '') {
-		static $cfr;
-		if (empty($cfr)) {
-			$cfr = cf_revisions::get_instance();
-		} 
-		return $cfr->register($postmeta_key, $display_func);
+
+define( 'CF_REVISIONS_DEBUG', true );
+
+/**
+ *
+ *
+ * @since 1.0.0
+ */
+function cfrm_register_metadata( $postmeta_key, $display_func = '' ) {
+	return cfrm_register( $postmeta_key, $display_func );
+}
+
+/**
+ *
+ *
+ * @since 2.0.0
+ */
+function cfrm_register( $postmeta_key, $display_func = '' ) {
+	if ( ! in_array( $postmeta_key, $this->postmeta_keys, true ) ) {
```

Comment:
This function will fail because of the use of `$this->` outside of a class context.